### PR TITLE
Restored ESP compatibility

### PR DESCRIPTION
**Pull request by dafes**
- Added ICACHE_RAM_ATTR before isr functions to restore esp compatibility:
[#6127](https://github.com/esp8266/Arduino/issues/6127)
- Removed some trailing whitespaces

### CODE REVIEW

**Comment by ladyada** [comment]
hi can you try/test this on non-ESP as well so we know it didnt break something with AVR/ARM

**Comment by dafes** [comment]
> hi can you try/test this on non-ESP as well so we know it didnt break something with AVR/ARM

hmm you are right... looks like this now breaks the avr stuff:
```
error: 'ICACHE_RAM_ATTR' does not name a type

 ICACHE_RAM_ATTR void RH_CC110::isr2()

 ^

exit status 1
Error compiling for board Arduino/Genuino Uno.
```
I will think about an fix for this and update the PR

**Comment by dafes** [comment]
Added an verify for [ESP8266 Buildflag](https://github.com/esp8266/Arduino/pull/26)

File: RH_CC110.cpp [hunk 1]
```diff
@@ -189,17 +189,29 @@ void RH_CC110::handleInterrupt()
...
+ICACHE_RAM_ATTR void RH_CC110::isr1()
+#else
 void RH_CC110::isr1()
+#endif
 {
     if (_deviceForInterrupt[1])
 	_deviceForInterrupt[1]->handleInterrupt();
 }
+#if defined(ESP8266)
+ICACHE_RAM_ATTR void RH_CC110::isr2()
```
Comment (ladyada) [inline RH_CC110.cpp]: you just need to ifdef the ICACHE word right? or you can #define ICACHE_RAM_ATTR for non-ESP8266 parts to nothing?

**Comment by gojimmypi** [comment]
perhaps something like this:

```
#if (RH_PLATFORM == RH_PLATFORM_ESP8266)
    // interrupt handler and related code must be in RAM on ESP8266,
    // according to issue #46.
    #define INTERRUPT_ATTR ICACHE_RAM_ATTR
#else
    #define INTERRUPT_ATTR
#endif
```
and use it like this:

```
// Read the RX data input pin, taking into account platform type and inversion.
bool INTERRUPT_ATTR RH_ASK::readRx()
{
    bool value;
#if (RH_PLATFORM == RH_PLATFORM_GENERIC_AVR8)
    value = ((RH_ASK_RX_PORT & (1<<RH_ASK_RX_PIN)) ? 1 : 0);
#else
    value = digitalRead(_rxPin);
#endif
    return value ^ _rxInverted;
}
```

**Comment by dafes** [comment]
Updated the PR to reflect the desired changes. It compiles under Arduino and works with my ESP8266

**Pull request merged**